### ISSUE TITLE
Can’t mount by NFS after upgrade vagrant to 2.3.1

### ISSUE THREAD

**Issue by chan15** [hashicorp/vagrant#12956]
#### Windows version
10 HOME 21H2

I used to mount the disk by NFS, after upgrading to **2.3.1**, `vagrant up` will be stuck for a while and eventually fail, but if downgraded to **2.2.19** NFS mounting will work.

#### Installed plugin
```
vagrant-winnfsd (1.4.0, global)
  - Version Constraint: > 0
 ```

```bash
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3,udp,actimeo=1,nolock 192.168.10.1:/D/projects /home/vagrant/code

Stdout from the command:



Stderr from the command:

mount.nfs: Connection timed out
```

**Comment by KuiKui** [comment]
Hi,

Same problem, but I can't rollback Vagrant due to the bug of Private Network Interface with MacOS 13 (see #12967), so I'm stuck to version 2.3.3

Is there a solution ? Even manual one ?
It was working like a charm before I'm forced to update to the 2.2.3

Thanks.

### Debug output

```
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3,udp,nolock,actimeo=1,fsc 192.168.56.1:/Users/xxx/workspace/app /vagrant

Stdout from the command:

Stderr from the command:
mount.nfs: Connection timed out
```

### Reproduction information 

#### Vagrant version

2.3.3 (no plugin installed)

#### Host operating system

MacOS 13.0.1 (`nfsd` has full access to the HD)

#### Guest operating system

Debian 10.8

#### Steps to reproduce

1. `vagrant up`

#### Vagrantfile

```ruby
Vagrant.configure('2') do |config|
    # Vagrant Box selection
    config.vm.box = "company/custom-box"
    config.vm.box_version = "0.5.5"

    # Network configuration
    config.vm.network 'private_network', , type: 'dhcp'

    # Sync configuration
    config.vm.synced_folder '.', '/vagrant', type: 'nfs', mount_options: ['nolock', 'actimeo=1', 'fsc']

    ...
end
```

**Comment by chan15** [comment]
No further progress, I just used the basic sync method, the execution speed inside the VM machine is way slower than NFS, and I guess they won't solve this issue.

**Comment by chrisroberts** [comment]
@KuiKui Your issue is different than the issue described here. It will be fixed by #13004 and a workaround for it is to define an IP for your private network (something like `config.vm.network :private_network, ip: '192.168.61.2'`)

@chan15 I've just spun up a basic ubuntu bionic64 guest on a windows 11 host using the latest Vagrant release with the vagrant-winnfsd plugin and it mounted just fine. Would you please provide a minimal Vagrantfile which reproduces the issue you are encountering and a gist of the debug log so I can investigate further. Thanks!

**Comment by chan15** [comment]
@chrisroberts 

**Software Version**

- Vagrant 2.3.3
- Virtualbox 7.0.2 r154219 (Qt5.15.2
- Windows 10 22H2
- vagrant-winnfsd (1.4.0, global)

**Vagrantfile**

```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/bionic64"
  config.vm.network "private_network", ip: "10.10.10.18"
  config.vm.synced_folder "./data", "/data", type: "nfs"

  config.vm.provider "virtualbox" do |vb|
    vb.memory = "1024"
  end
end
```

**Result**

Executed by Powershell 7.3 (administrator mode)

```bash
vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'ubuntu/bionic64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'ubuntu/bionic64' version '20220528.0.0' is up to date...
==> default: A newer version of the box 'ubuntu/bionic64' for provider 'virtualbox' is
==> default: available! You currently have version '20220528.0.0'. The latest is version
==> default: '20221102.0.0'. Run `vagrant box update` to update.
==> default: Setting the name of the VM: ubuntu18_default_1669083915568_37165
==> default: Fixed port collision for 22 => 2222. Now on port 2201.
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 (guest) => 2201 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2201
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection reset. Retrying...
    default: Warning: Connection aborted. Retrying...
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default:
    default: Inserting generated public key within guest...
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: The guest additions on this VM do not match the installed version of
    default: VirtualBox! In most cases this is fine, but in rare cases it can
    default: prevent things such as shared folders from working properly. If you see
    default: shared folder errors, please make sure the guest additions within the
    default: virtual machine match the version of VirtualBox you have installed on
    default: your host and reload your VM.
    default:
    default: Guest Additions Version: 5.2.42
    default: VirtualBox Version: 7.0
==> default: Configuring and enabling network interfaces...
==> default: Installing NFS client...
==> default: Exporting NFS shared folders...
==> default: Preparing to edit nfs mounting file.
[NFS] Status: running
==> default: Mounting NFS shared folders...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o vers=3,udp,vers=3,udp,nolock 10.10.10.1:/D/vagrant/ubuntu18/data /data

Stdout from the command:



Stderr from the command:

mount.nfs: Connection timed out
```

And it works if downgraded to the former version.

**Comment by pmhausen** [comment]
I found that with Mac OS 12.6.1, Virtualbox 7.0.4 and Vagrant 2.3.3 I have the same problem.

I did a clean boot of my Mac with all host-only networks removed from virtual box, private_network set to `type: 'dhcp'`.
I found that inside the guest Vagrant tries to mount from 192.168.56.1 while the bridge100 interface that was automatically created on the Mac had 192.168.56.3.

Possibly the cause?

**Comment by soapy1** [comment]
I can reproduce this problem using Vagrant 2.3.1 - 2.3.3, but seems to be fixed with 2.3.4 🙌 Can you confirm that upgrading Vagrant to 2.3.4 resolves the issue?

**Comment by KuiKui** [comment]
> I can reproduce this problem using Vagrant 2.3.1 - 2.3.3, but seems to be fixed with 2.3.4 🙌 Can you confirm that upgrading Vagrant to 2.3.4 resolves the issue?

Yes, it's works for me with vagrant 2.3.4 👍

**Comment by pmhausen** [comment]
Yes, NFS is working again.
